File: scrudbeans-integration-tests-java/src/main/java/myjavapackage/model/DiscountCode.java
package myjavapackage.model;

import com.github.manosbatsis.scrudbeans.api.mdd.annotation.model.ScrudBean;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;

/**
 * Sample entity model to test validation of non-null or unique @Column constraints
 */
@Entity
@Table(name = "discount_code")
@ScrudBean
@Schema(name = "DiscountCode", description = "A model representing an discount code")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DiscountCode {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column(nullable = false, unique = true)
    @Schema(description = "The discount code", required = true)
    private String code;

    @NotNull
    @Column(nullable = false)
    @Schema(description = "The discount percentage", required = true)
    private Integer percentage;
}

File: scrudbeans-integration-tests-kotlin/src/main/kotlin/mykotlinpackage/model/DiscountCode.kt
package mykotlinpackage.model

import com.github.manosbatsis.scrudbeans.api.annotation.model.ScrudBean
import com.github.manosbatsis.scrudbeans.model.BaseEntity
import io.swagger.v3.oas.annotations.media.Schema
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType.IDENTITY
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.validation.constraints.NotNull

/**
 * Sample entity model to test validation of non-null or unique @field:Column constraints
 */
@Entity
@Table(name = "discount_code")
@ScrudBean
@Schema(name = "DiscountCode", description = "A model representing an discount code")
data class DiscountCode(
    @field:Id
    @field:GeneratedValue(strategy = IDENTITY)
    override var id: Long = 0,

    @field:NotNull
    @field:Column(nullable = false, unique = true, updatable = false)
    @field:Schema(title = "The discount code", required = true)
    var code: String? = null,

    @field:NotNull
    @field:Column(nullable = false)
    @field:Schema(title = "The discount percentage", required = true)
    var percentage: Int? = null,
) : BaseEntity<Long>
